src/NLog/LayoutRenderers/Wrappers/WrapperLayoutRendererBase.cs

```diff
@@ -49,14 +52,29 @@ namespace NLog.LayoutRenderers.Wrappers
     /// </example>
     public abstract class WrapperLayoutRendererBase : LayoutRenderer
     {
+        private Layout _inner;
+
         /// <summary>
         /// Gets or sets the wrapped layout.
         /// 
         /// [DefaultParameter] so Inner: is not required if it's the first
         /// </summary>
         /// <docgen category='Transformation Options' order='10' />
         [DefaultParameter]
-        public Layout Inner { get; set; }
+        public Layout Inner
+        {
+            get => _inner;
+            set
+            {
+                _inner = value;
+                InnerChanged?.Invoke(this, EventArgs.Empty);
+            }
+        }
+
+        /// <summary>
+        /// Notify when <see cref="Inner"/> has been changed
+        /// </summary>
+        protected event EventHandler InnerChanged;
```

Comment:
For performance, but maybe overdesign

Reply:
Alternative one could use ReferenceEquals in the specialized class. And if the cached parsing points to a different Layout, then refresh the cached parsing.